### PR TITLE
feat(api-client): decouple the workspace store from the router

### PR DESCRIPTION
**Pull request by amritk**
This PR has all of the file changes to support #3931. Its just split up for easier digestion. You can see the actual changes there but test this one

### CODE REVIEW

File: packages/api-client/playground/web/main.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { createApiClientWeb } from '@/layouts/Web'
 
 createApiClientWeb(document.getElementById('scalar-client'), {
-  proxyUrl: 'https://proxy.scalar.com',
+  proxyUrl: 'https://proxy.mcdonalds.com',
```
Comment (marclave) [inline packages/api-client/playground/web/main.ts:4]: image

me

Comment (amritk) [reply packages/api-client/playground/web/main.ts]: 😆

**Comment by hwkr** [comment]
Looks good! I left a couple comments in the other PR. Also gave this a test but tbh not exactly sure how to test all of it...

I did try the oauth box and that appears to have some UI bugs (which are also on main), so Cam or I should fix that... (https://github.com/scalar/scalar/issues/3934)

https://github.com/user-attachments/assets/e575db94-39fe-4f2c-9089-0168800635ac

**Review by hanspagel — APPROVED** [review]
Love that we separate them! I don’t understand enough to have an opinion on the code changes, but I tested the modal, the web client and the Electron app and they all work perfectly fine. 🌟

**Comment by hanspagel** [comment]
Pls add a changeset before merging

**Comment by amritk** [comment]
@hwkr don't worry about fixing auth in references. These PR's are working up to removing it!

**Pull request merged**